### PR TITLE
 Moves views to MacPlatform for a better reuse

### PR DESCRIPTION
**Pull request by netonjm**
This was a @iainx suggestion in:
https://github.com/mono/monodevelop/pull/6324#discussion_r228878172

 I created a new PR to ensure I understood correctly the idea

### CODE REVIEW

File: main/src/addins/MacPlatform/AssemblyInfo.cs [hunk 1]
```diff
@@ -8,3 +8,4 @@
 [assembly: AssemblyDescription ("Mac Platform Support for MonoDevelop")]
 [assembly: AssemblyVersion ("2.6")]
 [assembly: AssemblyCopyright ("MIT X11")]
+[assembly: InternalsVisibleTo ("MonoDevelop.DesignerSupport, PublicKey=002400000c800000940000000602000000240000525341310004000001000100e1290d741888d13312c0cd1f72bb843236573c80158a286f11bb98de5ee8acc3142c9c97b472684e521ae45125d7414558f2e70ac56504f3e8fe80830da2cdb1cda8504e8d196150d05a214609234694ec0ebf4b37fc7537e09d877c3e65000f7467fa3adb6e62c82b10ada1af4a83651556c7d949959817fed97480839dd39b")]
```
Comment (Therzok) [inline main/src/addins/MacPlatform/AssemblyInfo.cs]: Delete this and add it here: https://github.com/mono/monodevelop/blob/master/main/src/addins/MacPlatform/MacPlatform.csproj#L152

File: main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.csproj [hunk 2]
```diff
@@ -76,6 +76,11 @@
       <Name>Xwt</Name>
       <Private>False</Private>
     </ProjectReference>
+    <ProjectReference Include="..\MacPlatform\MacPlatform.csproj">
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.csproj]: Should only be done in Mac configs.

**Comment by Therzok** [comment]
Why move them in MacPlatform though? [Components.Mac](https://github.com/mono/monodevelop/tree/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Mac) is where shared components live.

**Comment by iainx** [comment]
Yeah, I didn't mean to move all the views to MacPlatform, I meant take the Keycodes and Modifier enums from https://github.com/mono/monodevelop/blob/359226f90d9fbb2c56cf1556cdb4172d0ecdf848/main/src/addins/MacPlatform/MacInterop/KeyCodes.cs#L33-L43 and put them into Components.Mac, and then add the KeyCodes you used to the KeyCodes enum so we have them all in the one place that can be shared between the various projects

File: main/src/addins/MacPlatform/AssemblyInfo.cs (hunk 1 above)
Comment (netonjm) [reply main/src/addins/MacPlatform/AssemblyInfo.cs]: woah! nice trick thanks!

**Comment by netonjm** [comment]
sorry I read wrong the comment! clossing this right now :)

File: main/src/addins/MacPlatform/AssemblyInfo.cs (hunk 1 above)
Comment (Therzok) [reply main/src/addins/MacPlatform/AssemblyInfo.cs]: Yup, it's a custom msbuild task we have:
https://github.com/mono/monodevelop/blob/master/main/msbuild/GenerateInternalsVisibleTo.targets

**Pull request closed without merging**